cmd/root.go

```diff
@@ -39,8 +41,14 @@ var (
 		Run: func(cmd *cobra.Command, args []string) {
 			config := cli.MustLoadConfigFile()
 			server := cli.GetServer(config)
-			server.Up()
-			utils.ShellOut(config.PostUp, "PostUp")
+			if e := server.Up(); e != nil {
+				fmt.Printf("error bringing up the network: %s\n", e)
+				error_encountered = true
+			}
+			if e := utils.ShellOut(config.PostUp, "PostUp"); e != nil {
+				fmt.Printf("error bringing up the network: %s\n", e)
+				error_encountered = true
```

Comment:
I would suggest rather than setting a flag like this, you can change the [`Run` ](https://github.com/naggie/dsnet/pull/75/files#diff-ab967ab1a2f3a1b769106eeb7bfe892ef0e81d1d27811fa15be08e6749feee1fR41) to `RunE` which is the same as `Run` but lets you return an `error`

Reply:
I didn't know about `RunE()`; that's cleaner. What I didn't want to do was change the behavior. With `RunE()` there are two options: either return when the first error is encountered (which changes the behavior, since the following `ShellOut()` would not be called), or save the error and return one or the other (probably the first). 

If this were mine, I'd return at the first error encountered, but this might affect downstream folks who are expecting dsnet to behave this way (error, but continue). Which would you prefer?

Reply:
@botto Also, as I was going through this I noticed many funcs don't return errors, but just bail out deep in the function. Would you mind, for clarity and consistency, if I changed (e.g.) `cli.Sync()` to return an error, and then pass that error back up through `RunE()` as well? (c.f. [cmd/root.go#102](https://github.com/naggie/dsnet/blob/5adadaf40d57ea6f8bfb81693419b09f7a5fc980/cmd/root.go#L91))